packages/@lwc/engine-core/src/framework/base-lightning-element.ts

```diff
@@ -268,6 +274,90 @@ function warnIfInvokedDuringConstruction(vm: VM, methodName: string) {
 LightningElement.prototype = {
     constructor: LightningElement,
 
+    get children() {
+        const vm = getAssociatedVM(this);
+        const {
+            elm,
+            renderer: { getChildren },
+        } = vm;
+
+        if (process.env.NODE_ENV !== 'production') {
+            warnIfInvokedDuringConstruction(vm, 'children');
+        }
+
+        return getChildren(elm);
+    },
```

Comment:
What do you think about creating a helper function for all those methods to avoid code duplication?

Reply:
I did this in a previous PR, but it removed TypeScript guards, so we decided on this pattern instead. 🙂  I'm happy to reconsider it, though; I don't consider type safety as important as avoiding this much duplication. Or maybe there is a way to make the type safety work with the deduplication.

Reply:
I manage to deduplicate while maintaining type safety (https://github.com/salesforce/lwc/pull/2517/commits/4c3cb37faf71b34eb5a838fc74d026d65480efbb). It could probably be deduplicated further (e.g. `getBoundingClientRect`, defining getters and methods in the same loop), but I'm not enough of a TypeScript wizard to figure it out. 🧙‍♂️